test/extensions/filters/network/thrift_proxy/conn_manager_test.cc

```diff
@@ -1925,6 +1936,127 @@ stat_prefix: test
   EXPECT_EQ(nullptr, route);
 }
 
+TEST_F(ThriftConnectionManagerTest, UnknownCluster) {
+  const std::string yaml = R"EOF(
+transport: FRAMED
+protocol: BINARY
+stat_prefix: test
+route_config:
+  name: "routes"
+  routes:
+    - match:
+        method_name: name
+      route:
+        cluster: cluster
+)EOF";
+
+  EXPECT_THROW_WITH_REGEX(initializeFilter(yaml), EnvoyException, "unknown thrift cluster");
```

Comment:
I think this assertion could be done in the UnknownCluster2.

Reply:
You mean this is covered by the next test case? That makes sense to remove this given the next test case is not complicated.

Reply:
I mean you can move the `EXPECT_THROW_WITH_REGEX(....` into the next case (and delete this case).
For that call the same config is ok what is prepared in the next case.